toolchain/docs/check.md

```diff
@@ -0,0 +1,608 @@
+# Check
+
+<!--
+Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+Exceptions. See /LICENSE for license information.
+SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+-->
+
+<!-- toc -->
+
+## Table of contents
+
+-   [Postorder processing](#postorder-processing)
+-   [Key IR concepts](#key-ir-concepts)
+-   [Parameters and arguments](#parameters-and-arguments)
+-   [SemIR textual format](#semir-textual-format)
+-   [Raw form](#raw-form)
+-   [Formatted IR](#formatted-ir)
+    -   [Instructions](#instructions)
+    -   [Top-level entities](#top-level-entities)
+-   [Core loop](#core-loop)
+-   [Node stack](#node-stack)
+-   [Delayed evaluation (not yet implemented)](#delayed-evaluation-not-yet-implemented)
+-   [Templates (not yet implemented)](#templates-not-yet-implemented)
+-   [Rewrites](#rewrites)
+-   [Types](#types)
+-   [Type printing (not yet implemented)](#type-printing-not-yet-implemented)
+-   [Expression categories](#expression-categories)
+    -   [ExprCategory::NotExpression](#exprcategorynotexpression)
+    -   [ExprCategory::Value](#exprcategoryvalue)
+    -   [ExprCategory::DurableReference and ExprCategory::EphemeralReference](#exprcategorydurablereference-and-exprcategoryephemeralreference)
+    -   [ExprCategory::Initializing](#exprcategoryinitializing)
+    -   [ExprCategory::Mixed](#exprcategorymixed)
+-   [Value bindings](#value-bindings)
+-   [Handling Parse::Tree errors (not yet implemented)](#handling-parsetree-errors-not-yet-implemented)
+
+<!-- tocstop -->
+
```

Comment:
I prefer to avoid redundancy due to the likely skew over time from putting identical content in multiple files. How about instead this approach which drops the summaries from README.md?

Also, I noticed "lower" was in the wrong file, so fixing that, while also giving the driver and lex their own files. They could probably have more detail anyways.